### PR TITLE
[New feature](statistics) Step1: Statistics collection framework

### PR DESCRIPTION
**Pull request by EmmyMiao87**
# Proposed changes

Framework code for statistics collection,
containing only the main data structures, no implementation details.
This pr will not affect any existing code
and users will not be able to create statistics job.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by EmmyMiao87** [comment]
Linked #6370

File: fe/fe-core/src/main/java/org/apache/doris/common/Config.java [hunk 1]
```diff
@@ -1582,4 +1582,23 @@
      */
     @ConfField
     public static boolean allow_replica_on_same_host = false;
+
+    // statistics
+    /*
+     * the max unfinished statistics job number
+     */
+    @ConfField(mutable = true, masterOnly = true)
```
[the remark this answers is not in the document]
Comment (EmmyMiao87) [reply fe/fe-core/src/main/java/org/apache/doris/common/Config.java:1590]: Sure

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
Compile failed

**Review by morningman — APPROVED** [review]

**Pull request merged**